### DISCUSSION TITLE
PC-BASIC release 2.0.4

### DISCUSSION THREAD

**Discussion by robhagemans** [robhagemans/pcbasic#154] [Announcements]
For desktop use, use one of the installers here. For Python use, install with

    pip install pcbasic

## Selected improvements
- Many bug fixes, thanks to all contributors and reporters!
- Improved default font

## Notes
A number of packaging issues remain unsolved. In particular:

- The MacOS app is broken at present and fails to launch. It is not included. MacOS users could fall back to the last known working version 2.0.2 or install through `pip`.
- The Windows package continues to rely on Python 2.7, due to unsolved issues with the cx_Freeze setup on Python 3.

Any help with these packaging issues would be greatly appreciated.

## SHA-256 hashes

8e51b66d1ec26bc476cf68bb21f6c38dc5dbd17f7eb5d01968ef478a3f8067de  PC-BASIC-2.0.4.msi
09339ed4e4354679ff869422e24ccf0576f9cce26c28d2897b9f22ab58d579f1  pcbasic-2.0.4-py2.py3-none-any.whl
d1260bc840c66b85186ccacc35ca28c387a6b8ee0cb6285789e35960584031ef  python3-pcbasic-2.0.4-1.noarch.rpm
e32c119ae72e32646b687299a0d4e44a5512c5468e208de4dfe16710a9081eda  python3-pcbasic_2.0.4_all.deb

<hr /><em>This discussion was created from the release <a href='https://github.com/robhagemans/pcbasic/releases/tag/'>PC-BASIC release 2.0.4</a>.</em>

**Comment by JoeSolomonKCMO** [comment]
I was just trying to find a place to say Thank You for making PC Basic! It works perfectly so far and I am so happy with it! 

I am 65 years old. I can remember discovering Basic for the first time in Summer or Fall of 1975 as a college student. I was introduced to it by another student who told me about a Star Trek game on the university computer. After that I bugged him every day until he went with me over to one of the terminal locations and got me logged in. The original Star Trek was cancelled in 1969, 6 years prior. After that fans lived off of the James Blish adaptation paperbacks which were very well done. So it was with quite a bit of enthusiasm that I sat down to play the game. It almost made me feel that I was at Sulu's navigation console! I was hooked! 

There were 3 versions of Star Trek games, and other games, but over time they all lacked something or another. So I bought David Ahl's Basic Programming book at the campus library and was I ever intrigued! I began by making small alterations. In a Golf game, I added a score card that printed out on the screen at the end of the game, including "Eagle" "Bogey" etc descriptors and ability to print it out on the campus printer.  In one Star Trek game vs Romulans that was almost impossible to understand much less win (probably written by a student?) I got so frustrated that I changed the energy and formula so if nothing else worked I could self destruct and take them all out with me! ha.

Currently, I am using PC-Basic to rework Super Star Trek. The version I have came with errors. There wasn't even a RANDOMIZE command so every game set up exactly the same. Fixed it. I'm going to rework the quirky Damage Routine that fixes things that aren't broke(!) and breaks things for no reason(!!) I am amazed at the talent and the obviously deep understanding of string manipulation of the original authors, but the way they wrote it makes it challenging to edit. Might have been their intention?  I figured out the command "Color 2" changes text to a nice green color. I 'think' that was the text color on the university screens, but might have been a lighter shade.  

![SST](https://user-images.githubusercontent.com/117715884/200507580-215db760-d9c9-470b-acdd-377c882fd71c.jpg)

Now I'm thinking about reworking vintage code (as a cleaned up original), and making new improved versions, as well as writing my own games. I did so much coding in school and later when I started buying computers like Commodore 64 and etc that I guess I'm not surprised how quickly Basic came back to me. Having a BLAST!

So thanks very much again, Rob Hageman!

**Reply by Godswordislove** [reply]
why cant i save my program in pcbasic to my pc?

**Reply by robhagemans** [reply]
Thanks for the feedback! Nice to see some BASIC projects being revived

**Comment by JoeSolomonKCMO** [comment]
Try using

SAVE"FILE"
Then search your computer to see where it saved it. Mine shows up in
documents folder.

Then  LOAD"FILE" to bring it back

Theres probably a way to change the folder to save to but I'm new with
PCBasic also, I don't know.

**Reply by robhagemans** [reply]
See here for where files go, and how to change it: https://robhagemans.github.io/pcbasic/doc/2.0/#mounting

**Comment by JoeSolomonKCMO** [comment]
Welcome. I'm now writing new games using PCBasic currently working on
College Football Dynasty 1.0. Info posted at below accounts:

twitter: @vintagebasiclab
patreon: https://www.patreon.com/user?u=82517868
